rsconnect/certificates.py

```diff
@@ -0,0 +1,37 @@
+from pathlib import Path
+
+BINARY_ENCODED_FILETYPES = [".cer", ".der"]
+TEXT_ENCODED_FILETYPES = [".ca-bundle", ".crt", ".key", ".pem"]
+
+
+def read_certificate_file(location: str):
+    """Reads a certificate file from disk.
+
+    The file type (suffix) is used to determine the file encoding.
+    Assumption are made based on standard SSL practices.
+
+    Files ending in '.cer' and '.der' are assumed DER (Distinguished
+    Encoding Rules) files encoded in binary format.
+
+    Files ending in '.ca-bundle', '.crt', '.key', and '.pem' are PEM
+    (Privacy Enhanced Mail) files encoded in plain-text format.
+    """
+
+    path = Path(location)
+    suffix = path.suffix
```

Comment:
Checking the file type by examining its magic number is a safer way of inference than checking the suffix. This especially pertains to security related files like certificates.

Reply:
Good idea. I'll look into this a bit. 

Do you happen to know of any documentation that shows how to accomplish this across each of the certificate types?

Reply:
We use [mimetypes](https://docs.python.org/3/library/mimetypes.html) to infer some file types. It may not have full coverage of certificates, but it's a good starting point.

Reply:
It doesn't appear that certificates contain standard header blocks, but PEM files and DER files both adhere to a standard. https://www.openssl.org/docs/man3.0/man1/openssl-format-options.html

I can add additional validation that inspects the contents to check expecatations based on this documentation. But, I'm not sure if that is necessary. 

@mmarchetti - do you have any additional insights into this?

Reply:
`mimetypes` determines types by file suffix, so it's probably equivalent to what we're doing. I don't think we need to inspect file contents - if an invalid certificate file is provided, it will fail with an SSL error when we try to connect.